data_tools/scheme.py

```diff
@@ -0,0 +1,100 @@
+import validators
+
+class Scheme(object):
+    scheme_id = None
+    name = None
+    version = None
+    codes = []
+    documentation = dict()
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return other.scheme_id == self.scheme_id & \
+            other.name == self.name & \
+            other.version == self.version & \
+            other.documentation == self.documentation & \
+            other.codes == self.codes
+    
+    def __neq__(self, other):
+        return not self.__eq__(other)
+
+
+    @staticmethod
+    def from_firebase_map(data):
+        scheme = Scheme()
+        scheme.scheme_id = validators.validate_string(data["SchemeID"])
+        scheme.name = validators.validate_string(data["Name"])
+        scheme.version = validators.validate_string(data["Version"])
+
+        for code_map in data["Codes"]:
+            code = Code.from_firebase_map(code_map)
+            assert code.code_id not in code_map.keys(), \
+                "Non-unique Code Id found in scheme: {}".format(code.code_id)
+            scheme.codes.append(code)
+
+        if "Documentation" in data.keys():
+            doc_map = data["Documentation"]
+            scheme.documentation["URI"] = validators.validate_string(doc_map["URI"])
+        
+        return scheme
+    
+    def to_firebase_map(self):
+        ret = {}
+        ret["SchemeID"] = self.scheme_id
+        ret["Name"] = self.name
+        ret["Version"] = self.version
+        ret["Codes"] = []
+        for code in self.codes:
+            ret["Codes"].append(code.to_firebase_map())
+
+        if len(documentation.items) > 0:
+            ret["Documentation"] = documentation
+        
+        return ret
+        
+
+class Code:
+    code_id = None
+    display_text = None
+    shortcut = None
+    numeric_value = -42
+    visible_in_coda = True
+    color = None
+
+    @staticmethod
+    def from_firebase_map(data):
+        code = Code()
+        code.code_id = validators.validate_string(data["CodeID"], "CodeID")
+        code.display_text = validators.validate_string(data["DisplayText"], "DisplayText")
+        if "ShortCut" in data.keys():
```

Comment:
typo: I think it should be "Shortcut" (lowercase c)

Reply:
Thanks!